Tasks/AzureTestPlanV0/getAutomatedTests.ts

```diff
@@ -0,0 +1,105 @@
+﻿import { Console } from "console";
+import tl = require('azure-pipelines-task-lib/task');
+import apim = require('azure-devops-node-api');
+import { WorkItemDetails, TestCase } from 'azure-devops-node-api/interfaces/TestPlanInterfaces';
+import { PagedList } from 'azure-devops-node-api/interfaces/common/VSSInterfaces';
+import TestPlanInterfaces = require('azure-devops-node-api/interfaces/TestPlanInterfaces');
+import VSSInterfaces = require('azure-devops-node-api/interfaces/common/VSSInterfaces');
+
+
+export interface TestPlanData {
+    testPointIds: number[];
+    listOfFQNOfTestCases: string[];
+}
+
+ const AutomatedTestName = "Microsoft.VSTS.TCM.AutomatedTestName";
+ const AutomatedTestStorage = "Microsoft.VSTS.TCM.AutomatedTestStorage";
+export async function getAutomatedTestData(testPlanId: number, testSuiteIds: number[], testConfigurationId: number): Promise<TestPlanData> {
+    
+    const testPlanData: TestPlanData = { testPointIds: [], listOfFQNOfTestCases: [] };
+    let token = null;
+
+    if (testPlanId == 0 || testConfigurationId == 0) {
+        return testPlanData;
+    }
+
+    for (const testSuiteId of testSuiteIds) {
+        if (testSuiteId === 0) {
+            continue;
+        }
+
+        let testCasesData: TestCase[] = [];
+
+        do {
+            try {
+                let testCasesResponse = await fetchTestPlanList(testPlanId, testSuiteId, testConfigurationId.toString(), token);
+
+                token = testCasesResponse.continuationToken;
+
+                for (let key in testCasesResponse) {
+                    if (testCasesResponse.hasOwnProperty(key)) {
+                        testCasesData.push(testCasesResponse[key]);
+                    }
+                }
+
+            } catch (error) {
+                tl.error("Error fetching test cases list:" + error);
+                token = undefined;
+            }
+        } while (token !== undefined);
+
+        if (testCasesData.length === 0) {
+            console.log(`No test cases for test suite ${testSuiteId}`);
+            continue;
+        }
+
+        testCasesData.forEach(testCase => {
+            let automatedTestName = '';
+            let automatedTestStorage = '';
+
+            for (const witField of testCase.workItem?.workItemFields || []) {
+                const parsedWitField = JSON.parse(JSON.stringify(witField)); // Deep copy for safety
+
+                if (parsedWitField[AutomatedTestName] !== undefined) {
+                    automatedTestName = parsedWitField[AutomatedTestName].toString();
+                    testPlanData.listOfFQNOfTestCases.push(automatedTestName);
```

Comment:
Do you want to have a check for empty here?

Reply:
It is already there on line 63